nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -582,6 +582,8 @@ in
             force = cfgZfs.forceImportRoot;
             prefix = "/sysroot";
           }) rootPools);
+          targets.zfs-import.wantedBy = [ "zfs.target" ];
+          targets.zfs.wantedBy = [ "initrd.target" ];
```

Comment:
Is zfs.target reached in case pools are slightly degraded, but still mountable?

Reply:
Hm, I suppose I changed `wantedBy = ... ["local-fs.target"]` to `requiredBy = ... ["zfs-import.target"]`. So if any pool fails to import, then `zfs-import.target` will fail. But that is only *wanted* by `zfs.target`, so that would still be reached. Also, our `zfs-import-${pool}.service` *does* attempt to import pools degraded if possible.

Reply:
I'm not sure about having a generic name like `zfs.target`, could have something more descriptive?

Reply:
@RaitoBezarius That's just the name of the upstream unit that we get from `systemd.packages = [ cfgZfs.package ];`. I don't think we can change it, for compatibility's sake.

Reply:
Got it

Reply:
@flokli Is there a problem with this?

Reply:
> @flokli Is there a problem with this?

I don't know, I mostly wanted to make sure we can still boot even with a degraded pool. You don't want your machine to be unbootable just because one disk doesn't come back after a reboot.

If you're certain this still works, feel free to mark this thread as resolved :-)

Reply:
Well the logic for handling a degraded pool is in the import service itself, so I don't think that concern has to do with the target ordering.